app/controllers/api/vcs.php

```diff
@@ -0,0 +1,1019 @@
+    ->label('sdk.method', 'getInstallation')
+    ->label('sdk.description', '/docs/references/vcs/get-installation.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_INSTALLATION)
+    ->param('installationId', '', new Text(256), 'Installation Id')
+    ->inject('response')
+    ->inject('project')
+    ->inject('dbForProject')
+    ->inject('dbForConsole')
+    ->action(function (string $installationId, Response $response, Document $project, Database $dbForProject, Database $dbForConsole) {
+        $installation = $dbForConsole->getDocument('vcsInstallations', $installationId);
+
+        if ($installation === false || $installation->isEmpty()) {
+            throw new Exception(Exception::INSTALLATION_NOT_FOUND);
+        }
+
+        if ($installation->getAttribute('projectInternalId') !== $project->getInternalId()) {
+            throw new Exception(Exception::INSTALLATION_NOT_FOUND);
+        }
+
+        $functions = Authorization::skip(fn () => $dbForProject->find('functions', [
+            Query::equal('vcsInstallationInternalId', [$installation->getInternalId()]),
+            Query::limit(APP_LIMIT_SUBQUERY)
+        ]));
+
+        $installation->setAttribute('functions', $functions);
+
+        $response->dynamic($installation, Response::MODEL_INSTALLATION);
+    });
+
+App::delete('/v1/vcs/installations/:installationId')
+    ->groups(['api', 'vcs'])
+    ->desc('Delete Installation')
+    ->label('scope', 'public')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'vcs')
+    ->label('sdk.method', 'deleteInstallation')
+    ->label('sdk.description', '/docs/references/vcs/delete-installation.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_NOCONTENT)
+    ->label('sdk.response.model', Response::MODEL_NONE)
+    ->param('installationId', '', new Text(256), 'Installation Id')
+    ->inject('response')
+    ->inject('project')
+    ->inject('dbForConsole')
+    ->inject('deletes')
+    ->action(function (string $vcsInstallationId, Response $response, Document $project, Database $dbForConsole, Delete $deletes) {
+
+        $installation = $dbForConsole->getDocument('vcsInstallations', $vcsInstallationId, [
+            Query::equal('projectInternalId', [$project->getInternalId()])
+        ]);
+
+        if ($installation->isEmpty()) {
+            throw new Exception(Exception::INSTALLATION_NOT_FOUND);
+        }
+
+        if (!$dbForConsole->deleteDocument('vcsInstallations', $installation->getId())) {
+            throw new Exception(Exception::GENERAL_SERVER_ERROR, 'Failed to remove installation from DB');
+        }
+
+        $deletes
+            ->setType(DELETE_TYPE_DOCUMENT)
+            ->setDocument($installation);
+
+        $response->noContent();
+    });
+
+App::get('/v1/vcs/github/installations/:installationId/repositories/:repositoryId/detection')
```

Comment:
Also, @vermakhushboo even if this is needed, we should order our routes accordingly and group similar entities together. In this case, this route should've been grouped with the rest of the `/v1/vcs/github` routes.

Reply:
1. This endpoint might be used to detect function settings like build and install commands in the future for hosting. We should keep it separate.
2. Since `getRepository` is called every time a user selects a repo from function settings page, if we move this logic there, there would be extra calls to GitHub for detection. Do we want to do that?
3. I've moved it near the `listRepositories` endpoint

Reply:
We discussed this on meet and decided to keep it while introducing similar logic to listRepositories.